protocols/imap/src/main/java/org/apache/james/imap/encode/FetchResponseEncoder.java

```diff
@@ -290,6 +296,19 @@ private void encodeInternalDate(ImapResponseComposer composer, FetchResponse fet
         }
     }
 
+    private void encodeSaveDate(ImapResponseComposer composer, FetchResponse fetchResponse) throws IOException {
+        final Optional<Date> saveDate = fetchResponse.getSaveDate();
+        if (isSaveDateFetched(saveDate)) {
```

Comment:
We have Optional and still check null here?

Reply:
I needed a way to propagate whether the save date was fetched. With the current `FetchReponse`, the normal way is to use `null` to annotate it is not fetched.

However, the save date value is in Optional form, so it actually could make this a bit confusing, that is why I named `isSaveDateFetched(saveDate)` for more readability.

Better idea is welcome.